workspaces-tests/tests/upgrade.rs

```diff
@@ -0,0 +1,206 @@
+        .args_json(json!({}))
+        .unwrap()
+        .transact()
+        .await
+        .unwrap()
+        .json::<u32>()
+        .unwrap();
+
+    assert_eq!(val, 1);
+
+    let request_id = alice
+        .call(&worker, contract.id(), "upgrade_all")
+        .max_gas()
+        .args(NEW_WASM.to_vec())
+        .transact()
+        .await
+        .unwrap();
+
+    let new_val = alice
+        .call(&worker, contract.id(), "get_bar")
+        .args_json(json!({}))
+        .unwrap()
+        .transact()
+        .await
+        .unwrap()
+        .json::<u64>()
+        .unwrap();
+
+    assert_eq!(new_val, 1);
+
+    alice
+        .call(&worker, contract.id(), "decrement_bar")
+        .args_json(json!({}))
+        .unwrap()
+        .transact()
+        .await
+        .unwrap();
+
+    let end_val = alice
+        .call(&worker, contract.id(), "get_bar")
+        .args_json(json!({}))
+        .unwrap()
+        .transact()
+        .await
+        .unwrap()
+        .json::<u32>()
+        .unwrap();
+
+    assert_eq!(end_val, 0);
+}
+
+#[tokio::test]
+#[should_panic = "called `Result::unwrap()` on an `Err` value: Action #1: CompilationError(PrepareError(Deserialization))"]
+async fn upgrade_failure_blank_wasm() {
+    let Setup {
+        worker,
+        contract,
+        accounts,
+    } = setup(1, WASM).await;
+
+    let alice = &accounts[0];
+
+    alice
+        .call(&worker, contract.id(), "upgrade_all")
+        .max_gas()
+        .args(vec![])
+        .transact()
+        .await
+        .unwrap();
+}
+
+#[tokio::test]
+#[should_panic = "called `Result::unwrap()` on an `Err` value: Action #0: MethodResolveError(MethodNotFound)"]
+async fn upgrade_failure_no_upgrade() {
+    let Setup {
+        worker,
+        contract,
+        accounts,
+    } = setup(1, BAD_WASM).await;
+
+    let alice = &accounts[0];
+
+    alice
+        .call(&worker, contract.id(), "upgrade_all")
+        .max_gas()
+        .args(vec![])
+        .transact()
+        .await
+        .unwrap();
+}
+
+#[tokio::test]
+#[should_panic = "called `Result::unwrap()` on an `Err` value: Action #0: MethodResolveError(MethodNotFound)"]
+async fn upgrade_failure_random_wasm() {
+    let Setup {
+        worker,
+        contract,
+        accounts,
+    } = setup(1, RANDOM_WASM).await;
+
+    let alice = &accounts[0];
+
+    alice
+        .call(&worker, contract.id(), "upgrade_all")
+        .max_gas()
+        .args(vec![])
+        .transact()
+        .await
+        .unwrap();
+}
+
+#[tokio::test]
+#[should_panic = "called `Result::unwrap()` on an `Err` value: Action #0: ExecutionError(\"Smart contract panicked: Owner only\")"]
+async fn upgrade_failure_not_owner() {
+    let Setup {
+        worker,
+        contract,
+        accounts,
+    } = setup(2, WASM).await;
+
+    let alice = &accounts[0];
+    let bob: &Account = &accounts[1];
+
+    bob.call(&worker, contract.id(), "upgrade_all")
+        .max_gas()
+        .args(vec![])
```

Comment:
If we're just passing an empty Vec to `args`, can we just leave off the call entirely?

Reply:
done